src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,41 @@
-import /* we need our action types here*/ "../actions";
+import {
+  /* we need our action types here*/ FETCHING,
+  SUCCESS,
+  FAILURE
+} from "../actions";
+
 const initialState = {
-  characters: []
+  characters: [],
+  fetching: false,
+  error: null
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+    case FETCHING:
+      return {
+        ...state,
+        fetching: true
+      };
+
+    case SUCCESS:
+      console.log(action.payload);
+      return {
+        ...state,
+        characters: [...state.characters, ...action.payload],
```

Comment:
this is a good idea of what you need to do, but you have to know where to use it.
looking at your axios request your fetching the entire array of characters and setting that to your list, so you dont want to keep the current list you have, or you might get duplicates in your list. 

it just takes some practice and a few good console logs to know exactly what your actions/functions are outputting when your coding.
```suggestion
        characters: [ ...action.payload],
```